cls documentation (#575)
kyma-project/control-plane: docs/kyma-environment-broker/03-12-cls-account-setup.md

---
title: Cloud Logging Service Account setup
type: Details
---

The CLS instances are created inside a subaccount inside the Kyma global account: When the customer provisions the first SKR instance (SKR 1) for their SKR global account, one CLS instance is created inside the Kyma global account.

The following image explains the account setup for SKR and CLS instances:

![CLS Account diagram](./assets/cls-acc.jpeg)

The diagram illustrates the following: 
1. When a customer with global account X provisions the first SKR (SKR 1), a CLS instance (CLS for customer X) is created in the CLS subaccount that is part of the Kyma global account. 

> 2. After provisioning a CLS instance, a binding is created, and the credentials are passed over to SRK 1.  

Reviewer: ```suggestion
2. After provisioning a CLS instance, a binding is created, and the credentials are passed over to SRK 1. 
 
```

3. With this setup, the logs are pushed to the CLS instance.

For any subsequent provisioning of SKR (for example, SKR X-2) within the same global account, no new CLS instance is provisioned. Instead, the existing CLS instance is used by creating a new binding, which creates a new set of credentials to access the CLS instance. The new set of credentials is passed over to SKR X-2, so that it can push logs to CLS instance.

> To access Kibana one needs to authenticate against an ID provider based on SAML2 protocol

Reviewer: who is one? The user? 
```suggestion
To access Kibana, one must authenticate against an ID provider based on SAML2 protocol.
```
("must" is preferred over "need to")

> The information about the CLS integration with KEB can be found [here](./02-03-cls-integration.md)

Reviewer: ```suggestion
Learn more about the CLS integration with KEB under [Cloud Logging Service Integration](./02-03-cls-integration.md).
```

---
title: Cloud Logging Service Account setup
type: Details
---

The CLS instances are created inside a subaccount inside the Kyma global account: When the customer provisions the first SKR instance (SKR 1) for their SKR global account, one CLS instance is created inside the Kyma global account.

The following image explains the account setup for SKR and CLS instances:

> ![CLS Account diagram](./assets/cls-acc.jpeg)

Reviewer 2: Same comment about the diagram above

The diagram illustrates the following: 

> 1. When a customer with global account X provisions the first SKR (SKR 1), a CLS instance (CLS for customer X) is created in the CLS subaccount that is part of the Kyma global account. 

Reviewer 2: The list doesn't seem to be formatted properly in the preview: 

<img width="1547" alt="Screen Shot 2021-03-15 at 16 27 18" src="https://user-images.githubusercontent.com/49867570/111178415-68422800-85ab-11eb-9447-be1d305960e5.png">

> The diagram illustrates the following: 
> 1. When a customer with global account X provisions the first SKR (SKR X-1), a CLS instance (CLS for customer X) is created in the CLS subaccount that is part of the Kyma global account. 

Reviewer 2: ```suggestion
The diagram illustrates the following: 

1. When a customer with global account X provisions the first SKR (SKR X-1), a CLS instance (CLS for customer X) is created in the CLS subaccount that is part of the Kyma global account. 
```

> 2. After provisioning a CLS instance, a binding is created, and the credentials are passed over to SRK 1. 

Reviewer 2: ```suggestion
2. After provisioning a CLS instance, a binding is created, and the credentials are passed over to SKR 1. 
```

 
3. With this setup, the logs are pushed to the CLS instance.

For any subsequent provisioning of SKR (for example, SKR X-2) within the same global account, no new CLS instance is provisioned. Instead, the existing CLS instance is used by creating a new binding, which creates a new set of credentials to access the CLS instance. The new set of credentials is passed over to SKR X-2, so that it can push logs to CLS instance.

> To access Kibana, one must authenticate against an ID provider based on SAML2 protocol.

Reviewer 2: ```suggestion
To access Kibana, you must authenticate against an ID provider based on SAML2 protocol.
```
